src/components/AnnouncementModal/index.tsx

```diff
@@ -0,0 +1,119 @@
+import type React from 'react';
+import { Modal } from '@components/Modal/Modal';
+import { useEffect, useState } from 'react';
+import { Text } from '@components/Modal/components/Text';
+import { Button } from '@components/Modal/components/Button';
+import { getItem, setItem, websiteKey } from '@utils/storage';
+import settings from '@base/settings.json';
+import { Box } from '@components/Modal/components/Box';
+
+// 5  seconds
+const SHOW_MODAL_DELAY = 5 * 1000;
+
+type StorageValue = {
+  genericModalDismissed: boolean;
+  pathModalsDismissed: Record<string, boolean>;
+};
+
+const storageKey = websiteKey;
+
+// display modal again in 7 days
+const storageExpiresInDays = 7 as const;
+
+const AnnouncementModal: React.FC<{ pathname: string }> = ({ pathname }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const isGenericModal = settings.site.announcementModal.generic.visible;
+  const routeSpecificModal = !isGenericModal
+    ? settings.site.announcementModal.perPath.find(
+        (el) => pathname && el.path.includes(pathname),
+      )
+    : null;
+  const shouldOpenModal = isGenericModal || routeSpecificModal?.visible;
+  const targetModal = isGenericModal
+    ? settings.site.announcementModal.generic
+    : routeSpecificModal;
+
+  const saveModalDismissed = () => {
+    const existingStorage = getItem<StorageValue>(storageKey) || {
+      genericModalDismissed: false,
+      pathModalsDismissed: {},
+    };
+
+    if (isGenericModal) {
+      existingStorage.genericModalDismissed = true;
+    } else if (routeSpecificModal) {
+      existingStorage.pathModalsDismissed[
+        routeSpecificModal?.id || routeSpecificModal.path
+      ] = true;
+    }
+
+    setItem(storageKey, existingStorage, storageExpiresInDays);
+  };
+
+  const hasBeenDismissed = (): boolean => {
+    if (typeof window === 'undefined') return false;
```

Comment:
https://github.com/fleek-platform/website/blob/295348c90b80f2d93f1ab8e8c86d5425b46b743e/src/utils/common.ts#L2

```suggestion
    if (isServer) return false;
```